js/handlers.js

```diff
@@ -73,8 +82,21 @@ weechat.factory('handlers', ['$rootScope', '$log', 'models', 'plugins', 'notific
                 extra_date_format.year = "numeric";
             }
             content += " (";
-            content += new_date.toLocaleDateString(window.navigator.language,
-                                                   extra_date_format);
+            if ($rootScope.supports_formatting_date) {
+              content += new_date.toLocaleDateString(window.navigator.language,
+                                                     extra_date_format);
+            } else {
+              // ew ew not more gross code
+              var month_to_word = [
+                "January", "February", "March", "April",
+                "May", "June", "July", "August",
+                "September", "October", "November", "December"];
+              content += month_to_word[new_date.getMonth()] + " " +
+                new_date.getDate().toString();
+              content += (extra_date_format.year === "numeric") ?
```

Comment:
this would probably be better expressed as a normal condition, without the ternary operator.

Also, there's no need to make lines *that* short as those above and these